fastlane/Fastfile

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require './Constants/Constants'
+require './Managers/VersioningManager'
+require './Managers/BuildManager'
+require './Managers/SymbolManager'
+require './Managers/DistributionManager'
+
+versioning_manager = VersioningManager.new(
+  fastlane: self,
+  project_path: Constants.PROJECT_PATH,
+  main_target_name: Constants.MAIN_TARGET_NAME
+)
+
+builder = BuildManager.new(
+  fastlane: self,
+  scheme_name_staging: Constants.SCHEME_NAME_STAGING,
+  scheme_name_production: Constants.SCHEME_NAME_PRODUCTION,
+  scheme_name_appstore: Constants.SCHEME_NAME_APP_STORE,
+  bundle_identifier_staging: Constants.BUNDLE_ID_STAGING,
+  product_name_staging: Constants.PRODUCT_NAME_STAGING,
+  product_name_staging_test_flight: Constants.PRODUCT_NAME_STAGING_TEST_FLIGHT,
+  product_name_production: Constants.PRODUCT_NAME_PRODUCTION
+)
+
+distribution_manager = DistributionManager.new(
+  fastlane: self,
+  build_path: Constants.BUILD_PATH,
+  firebase_token: ENV['FIREBASE_CLI_TOKEN']
+)
+
+keychain_name = 'github_action_keychain'
+keychain_password = 'password'
+
+before_all do
+  ensure_bundle_exec
+end
+
+default_platform(:ios)
+
+platform :ios do
+  desc 'Sync development match signing'
+  lane :sync_development_signing do
+    match(type: 'development', readonly: true)
+  end
+
+  desc 'Sync AdHoc match signing'
+  lane :sync_adhoc_signing do
+    match(type: 'adhoc', readonly: true)
+  end
+
+  desc 'Sync Adhoc match signing on CI'
+  lane :sync_adhoc_signing_ci do
+    create_ci_keychain
+    match(
+        type: 'adhoc',
+        keychain_name: keychain_name,
+        keychain_password: keychain_password,
+        app_identifier: [Constants.BUNDLE_ID_STAGING],
+        readonly: true
+    )
+  end
```

Comment:
Should we have a class like `MatchManager`, and handle about running local or on ci with a flag, for example:
```rb
matchManger = MatchManger(fastlane:, is_ci: = false)
matchManger.sync_development_signing()
matchManger.sync_adhoc_signing()
```
What do you think?

Reply:
So we don't need to have 2 functions, one for local and one for CI